src/main/scala/com/databricks/labs/overwatch/ApiCallV2.scala

```diff
@@ -640,6 +652,123 @@ class ApiCallV2(apiEnv: ApiEnv) extends SparkSessionWrapper {
     }
   }
 
+  /**
+   * Function to make parallel API calls. Currently this functions supports only SqlQueryHistory and ClusterEvents
+   * @param endpoint
+   * @param jsonInput
+   * @param config
+   * @return
+   */
+  def makeParallelApiCalls(endpoint: String, jsonInput: Map[String, String], config: Config): String = {
+    var startValue = jsonInput.get("start_value").get.toLong
+    val endValue = jsonInput.get("end_value").get.toLong
```

Comment:
all the jsonInput may not contain start_value and end_value. This piece of code can be more generic. " jsonInput.get("start_value").get.toLong" this can cause null pointer exception.

Reply:
i guess we should move the part specific to the API to APIMETA.